src/function/list/list_append_function.cpp

```diff
@@ -28,23 +31,55 @@ struct ListAppend {
     }
 };
 
-static void validateArgumentType(const binder::expression_vector& arguments) {
-    if (ListType::getChildType(arguments[0]->dataType) != arguments[1]->getDataType()) {
-        throw BinderException(
-            ExceptionMessage::listFunctionIncompatibleChildrenType(ListAppendFunction::name,
-                arguments[0]->getDataType().toString(), arguments[1]->getDataType().toString()));
-    }
-}
-
 static std::unique_ptr<FunctionBindData> bindFunc(const ScalarBindFuncInput& input) {
-    validateArgumentType(input.arguments);
+
+    std::vector<LogicalType> types;
+    types.push_back(input.arguments[0]->getDataType().copy());
+    types.push_back(input.arguments[1]->getDataType().copy());
+
+    auto bothEmpty = [](std::vector<LogicalType>& types, LogicalType& targetType) {
+        targetType = types[1].copy();
+        if (targetType.getLogicalTypeID() == LogicalTypeID::ANY)
+            targetType = LogicalType(LogicalTypeID::INT64);
+    };
+
+    auto leftEmpty = bothEmpty;
```

Comment:
You don't have to define leftEmpty and rightEmpty, just pass bothEmpty to the function.